include/RAJA/exec-cuda/forall_cuda.hxx

```diff
@@ -201,7 +230,7 @@ RAJA_INLINE void forall(cuda_exec_async<BLOCK_SIZE>,
 
   RAJA_FT_BEGIN;
 
-  forall_cuda_kernel<<<gridSize, BLOCK_SIZE>>>(loop_body, begin, len);
+  forall_cuda_kernel<<<gridSize, BLOCK_SIZE,BLOCK_SIZE*sizeof(double)>>>(loop_body, begin, len);
```

Comment:
The value for len needs to be adjusted.  If the grid is 1027*2035*2097 then len is equal to that product.   I think this reduction has to be implemented directly in the context of the BoxSegment branch, but I see where this is going.  I'll take a stab at it on the BoxSegment branch.